api/window/src/main/java/quarks/window/Window.java

```diff
@@ -157,4 +158,10 @@ Licensed to the Apache Software Foundation (ASF) under one
      */
     Function<T, K> getKeyFunction();
 
+    /**
+     * Retrieves the partitions in the window.
+     * @return A map of the window's keys and partitions.
```

Comment:
Maybe you've also made adjustments but on lines 188 & 216 below in ConnectorStream, it's not synchronizing the partitions map when getting the partition of interest.  Q: would using/returning a synchronized map be better?